framework/sessions/__init__.py

```diff
@@ -143,9 +146,36 @@ def create_session(response, data=None):
 # NOTE: This gets attached in website.app.init_app to ensure correct callback
 # order
 def before_request():
+    from framework.auth import authenticate
+    from framework.auth.core import User
+    from framework.auth.cas import CasClient
+
+    # Central Authentication Server Ticket Validation and Authentication
+    ticket = request.args.get('ticket')
+    if ticket:
```

Comment:
This logic will need tests. Would be good to separate this into a function to make unit testing easier.